Import: Improve error and don't report problem if cache is cleared

david-allison opened the pull request:
The cache can be cleared, this would cause both a crash report dialog and a generic "failed to unzip" error.

## Fixes

Fixes #6491

## Approach
Catch exception and display a better message. Don't error report.

## How Has This Been Tested?

Tested on my phone, both by throwing the normal ENOENT and throwing a generic FileNotFound, the FileNotFound raises an exception report. the ENOENT doesn't

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

david-allison committed:
Import: Improve error if cache is cleared

Fixes 6491

mikehardy commented on AnkiDroid/src/main/res/values/02-strings.xml:
```diff
@@ -161,6 +161,7 @@
     <string name="import_log_failed_copy_to">Failed to copy apkg to %s</string>
     <string name="import_log_failed_validate">apkg failed validation</string>
     <string name="import_log_insufficient_space">There is not enough space to unzip the package. Needed %1$d, available %2$d</string>
+    <string name="import_log_file_cache_cleared">Temporary error importing file.\nPlease try again</string>
```
During my testing large decks for OOM/OOSpace I noted that the Android OS actively prompts you, fairly urgently that you are running out of space and suggests actions including clearing caches. The combination of that fail mode and a more pointed message here might connect things better and hand-hold the users?

```suggestion
    <string name="import_log_file_cache_cleared">Error importing file, likely a cache clear during processing.\nPlease try again</string>
```

david-allison-1 replied:
Oh! I didn't realise, if the user has a choice in the matter, then agreed.

mikehardy reviewed (changes requested):
Nice edge case catch - my suggested word change might be improved (I'm never certain on wordsmithing) but I definitely want the message to connect the user action to the failure so their chance of success goes up without any other info needed

mikehardy commented:
releng note: I'd also pick this to release-2.11 if we do anything there, nice contained UX improvement

david-allison committed:
NF: Improve Error message

Users may have a choice about cache clearing, this makes it better to explain it was a cache clear

Co-authored-by: Mike Hardy <github@mikehardy.net>

david-allison-1 commented:
Done. Could you squash-merge when Travis passes

mikehardy reviewed (approved):
LGTM - squash merge request noted
